pkg/transformer/kubernetes/kubernetes.go

```diff
@@ -702,40 +705,78 @@ func (k *Kubernetes) Undeploy(komposeObject kobject.KomposeObject, opt kobject.C
 	}
 
 	for _, v := range objects {
+		label := labels.SelectorFromSet(labels.Set(map[string]string{transformer.Selector: v.(meta.Object).GetName()}))
+		options := api.ListOptions{LabelSelector: label}
 		switch t := v.(type) {
 		case *extensions.Deployment:
 			//delete deployment
-			rpDeployment, err := kubectl.ReaperFor(extensions.Kind("Deployment"), client)
+			deployment, err := client.Deployments(namespace).List(options)
 			if err != nil {
 				return err
 			}
-			//FIXME: gracePeriod is nil
-			err = rpDeployment.Stop(namespace, t.Name, TIMEOUT*time.Second, nil)
-			if err != nil {
-				return err
+			komposeLabel := map[string]string{transformer.Selector: t.Name}
```

Comment:
You can move this outside of this switch. That you wont have to repeat same line in every case.